Feat: Added Pricing Component

SatyamVyas04 opened the pull request:
Added support for a pricing component, that displays plans in a well animated manner, using framer-motion
Here's the demo for the same. Let me know if there are any potential bugs or fallacies in the code 👍🏻 

https://github.com/user-attachments/assets/14aaec23-7894-4993-8fa7-dd73b7f67099

fixes #295

hari commented on animata/widget/pricing.tsx:
```diff
@@ -0,0 +1,181 @@
+      setPrice(`$${currentValue.toFixed(2)}`);
+
+      if (progress < 1) {
+        requestAnimationFrame(animatePrice);
```
You want to cancel the animation on unmount.

```
let rafId;
...

return () => {
  if (rafId) { 
      cancelAnimationFrame(cancelAnimationFrame);
    }
 }
 ...
 ```

hari commented on content/docs/widget/pricing.mdx:
```diff
@@ -0,0 +1,38 @@
+
+## Credits
+
+Built by [SatyamVyas04](https://github.com/SatyamVyas04)
```
Please add: `Inspired by [Nitish Khagwal](https://x.com/nitishkmrk)`

hari reviewed (changes requested):
@SatyamVyas04 Looking good. Here are a few things:

1. Please move this to `Sections` category, you will need to create a new one and rename the files as well text.
2. Add little more spacing.

SatyamVyas04 commented:
Working on the animation-unmount now 👍🏻

SatyamVyas04 commented:
This pull request introduces a new `Pricing` component and its associated documentation and stories. The changes include adding the component, creating Storybook stories for it, and updating the documentation to include installation and usage instructions.

### New `Pricing` Component:

* [`animata/section/pricing.tsx`](diffhunk://#diff-b28991ab69c65e4f21a865e3d72fd2448477c128c4f12fffec003ce687fed901R1-R223): Added a new `Pricing` component with customizable plans, width, outer radius, and padding. The component includes interactive elements for selecting billing cycles and plans, with animations for a better user experience.

### Storybook Stories:

* [`animata/section/pricing.stories.tsx`](diffhunk://#diff-e519edaa55797744e9f00e784b88f3b6244d37b423a574dca1707f6c2eb47aedR1-R56): Added Storybook stories for the `Pricing` component, including `Primary` and `CustomPlans` stories to demonstrate different configurations.

### Documentation Updates:

* [`content/docs/section/pricing.mdx`](diffhunk://#diff-1c329ec6c35d5ea841fcb54b4dd0d7e8a4bcfa1bad8acc636e34e255b44c2063R1-R39): Added new documentation for the `Pricing` component, including installation instructions, usage examples, and credits.

### Sidebar Navigation:

* [`config/docs.ts`](diffhunk://#diff-72eed20dacf77aeae0f24473af8500d68560c933a1ee432958b059bac1223650R112-R115): Updated the sidebar navigation to include a new "Section" category with links to the `Pricing` component documentation.
